### PR TITLE
feat: add CLI command for loading chunks to Read Buffer

### LINKED ISSUE #3267: Add ability to load Chunks to RUB via CLI
I want to be able to use the CLI to load chunks from Object Storage to the Read Buffer.

### PR DESCRIPTION
**Pull request by e-dard**
Closes #3267

This PR follows on the work that @tustvold has been doing to teach IOx to load chunks from object storage back into the Read Buffer.

### CLI command

```
$ influxdb_iox database chunk load NAME_OF_DATABASE CHUNK_ID_IN_BASE_16
```

### Testing

I have tested this locally. Start IOx with some persisted chunks and note:

```
influxdb_iox database query query_log "select * from system.chunks"

+--------------------------------------+---------------------+------------+-----------------+------------------+--------------+--------------------+-----------+---------------------+-----------------------------+-----------------------------+-------+
| id                                   | partition_key       | table_name | storage         | lifecycle_action | memory_bytes | object_store_bytes | row_count | time_of_last_access | time_of_first_write         | time_of_last_write          | order |
+--------------------------------------+---------------------+------------+-----------------+------------------+--------------+--------------------+-----------+---------------------+-----------------------------+-----------------------------+-------+
| 47f7d3a6-714e-4ffc-b8f6-0d144c7653d8 | 2021-11-30 10:00:00 | query_log  | ObjectStoreOnly |                  | 17791        | 31740929           | 409920    |                     | 2021-11-30T13:01:56.252683Z | 2021-11-30T13:02:26.684201Z | 1     |
| 5fafd0c1-0d0d-4798-87a0-6accb7d54ec1 | 2021-11-30 10:00:00 | query_log  | ObjectStoreOnly |                  | 33165        | 47253465           | 1022972   |                     | 2021-11-30T13:01:56.252683Z | 2021-11-30T13:02:54.778474Z | 1     |
| fb425523-c822-486c-b317-72a3ba7dbe57 | 2021-11-30 07:00:00 | query_log  | ObjectStoreOnly |                  | 28896        | 47654142           | 870285    |                     | 2021-11-30T13:00:22.754526Z | 2021-11-30T13:00:52.815853Z | 1     |
| 48642d71-1206-42a6-a5d8-5bc0faa752d4 | 2021-11-30 07:00:00 | query_log  | ObjectStoreOnly |                  | 21172        | 25529331           | 450712    |                     | 2021-11-30T13:00:53.403215Z | 2021-11-30T13:01:23.441424Z | 7     |
| 012e9c79-65c8-419c-8a60-c26fbe0bb913 | 2021-11-30 07:00:00 | query_log  | ObjectStoreOnly |                  | 8535         | 3034732            | 44656     |                     | 2021-11-30T13:01:23.879715Z | 2021-11-30T13:01:28.491953Z | 10    |
| 91be1c79-d7da-41f0-b52f-f8816fe1bc7c | 2021-11-30 09:00:00 | query_log  | ObjectStoreOnly |                  | 22652        | 50540440           | 655490    |                     | 2021-11-30T13:01:28.801387Z | 2021-11-30T13:02:01.241539Z | 1     |
| 4e265d94-1479-43f4-8699-7b35738c0d30 | 2021-11-30 09:00:00 | query_log  | ObjectStoreOnly |                  | 29555        | 36484629           | 741147    |                     | 2021-11-30T13:02:01.674413Z | 2021-11-30T13:02:24.800429Z | 6     |
| 2612a9fe-e823-432c-bc65-5c840fd21a08 | 2021-11-30 08:00:00 | query_log  | ObjectStoreOnly |                  | 24563        | 68062178           | 832540    |                     | 2021-11-30T13:00:56.568057Z | 2021-11-30T13:01:27.162309Z | 1     |
| b2152255-5ca8-455d-bbc2-59cdbfbf6818 | 2021-11-30 08:00:00 | query_log  | ObjectStoreOnly |                  | 23672        | 59992826           | 765536    |                     | 2021-11-30T13:00:56.568057Z | 2021-11-30T13:01:55.907730Z | 1     |
| 33837661-7501-460a-9b66-e64549b25b1c | 2021-11-30 05:00:00 | query_log  | ObjectStoreOnly |                  | 15202        | 18005733           | 214902    |                     | 2021-11-30T12:59:53.829897Z | 2021-11-30T13:00:22.399715Z | 1     |
| d2071087-31ab-4e69-90b8-b4e9d81c97fb | 2021-11-30 11:00:00 | query_log  | ObjectStoreOnly |                  | 18198        | 24025063           | 292940    |                     | 2021-11-30T13:02:25.133301Z | 2021-11-30T13:02:54.779559Z | 1     |
| 9b8985bb-0a1e-4c85-b4df-564d9e43f079 | 2021-11-30 06:00:00 | query_log  | ObjectStoreOnly |                  | 16665        | 23570552           | 275732    |                     | 2021-11-30T12:59:53.829199Z | 2021-11-30T13:00:56.207493Z | 1     |
| dddb3c40-15ca-4c02-9360-d865094ee092 | 2021-11-30 06:00:00 | query_log  | ObjectStoreOnly |                  | 28181        | 73027282           | 962017    |                     | 2021-11-30T12:59:53.829199Z | 2021-11-30T13:00:24.552314Z | 1     |
+--------------------------------------+---------------------+------------+-----------------+------------------+--------------+--------------------+-----------+---------------------+-----------------------------+-----------------------------+-------+
```

Then load a chunk:

```
$ influxdb_iox database chunk load query_log "5fafd0c1-0d0d-4798-87a0-6accb7d54ec1"
{
  "operation": {
    "name": "0",
    "metadata": {
      "typeUrl": "type.googleapis.com/influxdata.iox.management.v1.OperationMetadata",
      "value": "GAEgAZoBPQoJcXVlcnlfbG9nEhMyMDIxLTExLTMwIDEwOjAwOjAwGglxdWVyeV9sb2ciEF+v0MENDUeYh6BqzLfVTsE="
    }
  },
  "metadata": {
    "totalCount": "1",
    "pendingCount": "1",
    "loadReadBufferChunk": {
      "dbName": "query_log",
      "partitionKey": "2021-11-30 10:00:00",
      "tableName": "query_log",
      "chunkId": "X6/QwQ0NR5iHoGrMt9VOwQ=="
    }
  }
}%
```

Finally note:

```
+--------------------------------------+---------------------+------------+--------------------------+------------------+--------------+--------------------+-----------+-----------------------------+-----------------------------+-----------------------------+-------+
| id                                   | partition_key       | table_name | storage                  | lifecycle_action | memory_bytes | object_store_bytes | row_count | time_of_last_access         | time_of_first_write         | time_of_last_write          | order |
+--------------------------------------+---------------------+------------+--------------------------+------------------+--------------+--------------------+-----------+-----------------------------+-----------------------------+-----------------------------+-------+
| 47f7d3a6-714e-4ffc-b8f6-0d144c7653d8 | 2021-11-30 10:00:00 | query_log  | ObjectStoreOnly          |                  | 17791        | 31740929           | 409920    |                             | 2021-11-30T13:01:56.252683Z | 2021-11-30T13:02:26.684201Z | 1     |
| 5fafd0c1-0d0d-4798-87a0-6accb7d54ec1 | 2021-11-30 10:00:00 | query_log  | ReadBufferAndObjectStore |                  | 1638014408   | 47253465           | 1022972   | 2021-12-01T11:17:05.306057Z | 2021-11-30T13:01:56.252683Z | 2021-11-30T13:02:54.778474Z | 1     |
| fb425523-c822-486c-b317-72a3ba7dbe57 | 2021-11-30 07:00:00 | query_log  | ObjectStoreOnly          |                  | 28896        | 47654142           | 870285    |                             | 2021-11-30T13:00:22.754526Z | 2021-11-30T13:00:52.815853Z | 1     |
| 48642d71-1206-42a6-a5d8-5bc0faa752d4 | 2021-11-30 07:00:00 | query_log  | ObjectStoreOnly          |                  | 21172        | 25529331           | 450712    |                             | 2021-11-30T13:00:53.403215Z | 2021-11-30T13:01:23.441424Z | 7     |
| 012e9c79-65c8-419c-8a60-c26fbe0bb913 | 2021-11-30 07:00:00 | query_log  | ObjectStoreOnly          |                  | 8535         | 3034732            | 44656     |                             | 2021-11-30T13:01:23.879715Z | 2021-11-30T13:01:28.491953Z | 10    |
| 91be1c79-d7da-41f0-b52f-f8816fe1bc7c | 2021-11-30 09:00:00 | query_log  | ObjectStoreOnly          |                  | 22652        | 50540440           | 655490    |                             | 2021-11-30T13:01:28.801387Z | 2021-11-30T13:02:01.241539Z | 1     |
| 4e265d94-1479-43f4-8699-7b35738c0d30 | 2021-11-30 09:00:00 | query_log  | ObjectStoreOnly          |                  | 29555        | 36484629           | 741147    |                             | 2021-11-30T13:02:01.674413Z | 2021-11-30T13:02:24.800429Z | 6     |
| 2612a9fe-e823-432c-bc65-5c840fd21a08 | 2021-11-30 08:00:00 | query_log  | ObjectStoreOnly          |                  | 24563        | 68062178           | 832540    |                             | 2021-11-30T13:00:56.568057Z | 2021-11-30T13:01:27.162309Z | 1     |
| b2152255-5ca8-455d-bbc2-59cdbfbf6818 | 2021-11-30 08:00:00 | query_log  | ObjectStoreOnly          |                  | 23672        | 59992826           | 765536    |                             | 2021-11-30T13:00:56.568057Z | 2021-11-30T13:01:55.907730Z | 1     |
| 33837661-7501-460a-9b66-e64549b25b1c | 2021-11-30 05:00:00 | query_log  | ObjectStoreOnly          |                  | 15202        | 18005733           | 214902    |                             | 2021-11-30T12:59:53.829897Z | 2021-11-30T13:00:22.399715Z | 1     |
| d2071087-31ab-4e69-90b8-b4e9d81c97fb | 2021-11-30 11:00:00 | query_log  | ObjectStoreOnly          |                  | 18198        | 24025063           | 292940    |                             | 2021-11-30T13:02:25.133301Z | 2021-11-30T13:02:54.779559Z | 1     |
| 9b8985bb-0a1e-4c85-b4df-564d9e43f079 | 2021-11-30 06:00:00 | query_log  | ObjectStoreOnly          |                  | 16665        | 23570552           | 275732    |                             | 2021-11-30T12:59:53.829199Z | 2021-11-30T13:00:56.207493Z | 1     |
| dddb3c40-15ca-4c02-9360-d865094ee092 | 2021-11-30 06:00:00 | query_log  | ObjectStoreOnly          |                  | 28181        | 73027282           | 962017    |                             | 2021-11-30T12:59:53.829199Z | 2021-11-30T13:00:24.552314Z | 1     |
+--------------------------------------+---------------------+------------+--------------------------+------------------+--------------+--------------------+-----------+-----------------------------+-----------------------------+-----------------------------+-------+
```

Showing the chunk is now in both the object store and the Read Buffer.

### CODE REVIEW

File: data_types/src/chunk_metadata.rs [hunk 1]
```diff
@@ -254,13 +254,16 @@ impl From<ChunkId> for Bytes {
 }
 
 #[derive(Debug, Snafu)]
-pub enum BytesToChunkIdError {
+pub enum ChunkConversionError {
```
Comment (alamb) [inline data_types/src/chunk_metadata.rs:257]:
```suggestion
pub enum ChunkIdConversionError {
```

**Review by alamb — APPROVED** [review]
I wonder if an end to end test might be helpful to ensure this code isn't broken at some future point

**Comment by e-dard** [comment]
> I wonder if an end to end test might be helpful to ensure this code isn't broken at some future point

@alamb do we have a good way to test the CLI?

**Review by crepererum — CHANGES_REQUESTED** [review]
Please add a new test to <https://github.com/influxdata/influxdb_iox/blob/main/influxdb_iox/tests/end_to_end_cases/management_cli.rs>

**Comment by alamb** [comment]
> @alamb do we have a good way to test the CLI?

@crepererum suggested the [way](https://github.com/influxdata/influxdb_iox/pull/3268#pullrequestreview-820246889). There are plenty of examples of how to run a CLI test there 👍

**Review by crepererum — APPROVED** [review]
Please also apply @alamb s suggestion.

**Pull request merged**